libsolidity/ast/Types.h

```diff
@@ -1154,14 +1154,13 @@ class FunctionType: public Type
 	/// of the parameters to false.
 	TypePointer copyAndSetGasOrValue(bool _setGas, bool _setValue) const;
 
-	/// @returns a copy of this function type where all return parameters of dynamic size are
-	/// removed and the location of reference types is changed from CallData to Memory.
-	/// This is needed if external functions are called on other contracts, as they cannot return
-	/// dynamic values.
-	/// Returns empty shared pointer on a failure. Namely, if a bound function has no parameters.
+	/// @returns a copy of this function type where the location of reference types is changed
+	/// from CallData to Memory. This is the type that would be used when the function is
+	/// called, as opposed to the parameter types that are available inside the function body.
```

Comment:
I think I do not fully understand what that means.

Reply:
If you call an external function, you have to put the data into memory (and then issue the `call` opcode). If you are executing the function, then the arguments are in calldata instead of memory.